### ISSUE TITLE
FS.File.copy should allow modifying objects before saving

### ISSUE THREAD

**Issue by lukejagodzinski** [CollectionFS/Meteor-CollectionFS#491]
It should be possible to pass Mongo Modifier as a parameter of [FS.File.copy](https://github.com/CollectionFS/Meteor-cfs-file/blob/master/fsFile-server.js#L126) function to alter object just before saving it. The other solution could be one proposed by @aldeed that it could by just function that will be invoked before saving.

**Comment by adamgins** [comment]
Hi folks, just wondering if there was any progress on this?
If it's not possible to change the meta data before the copy, what's the best approach to modify it afterwards, pls? ie do I do the copy and then call some update function or just a straight mongodb `update`?

**Comment by CoolestNerdIII** [comment]
@adamgins I know this is kind of a late response, but I just encountered this and figured I would say what I was doing, figuring if anyone sees this and lets me know I did it wrong, it could help out people in the future.
My code looks a bit like this:

```
let newFile = original.copy();  

newFile.update({
  '$set': {
    'owner': Meteor.userId()
  }
});
```

I am doing this on the server, and it appears to work as intended.

**Comment by adamgins** [comment]
@CoolestNerdIII cheers. I am not using CollectionFS any longer.. especially as it now has the "deprecated" tag

**Comment by CoolestNerdIII** [comment]
Yeah I have been looking for another solution to migrate to. What do you use now if you don''t mind my asking?

**Comment by adamgins** [comment]
I ended going commercial filestack.com , I use it for www.buzzy.buzz
I did look at Slingshot https://github.com/CulturalMe/meteor-slingshot/ but needs some features that were not available.

**Comment by CoolestNerdIII** [comment]
Thanks I will check those out. Kinda wish this wasn't deprecated. Oh well

**Comment by adamgins** [comment]
yeah, I wish Meteor JS has had a set of core services for files, editors etc as part of the platform/framework. I know this is highly dependent on the UX approach chosen by each developer, but would be nice to have some core services that almost every app needs.

**Comment by CoolestNerdIII** [comment]
I agree 100%.
